security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -117,54 +113,17 @@ func (c *CitadelClient) CSRSign(csrPEM []byte, certValidTTLInSec int64) ([]strin
 	return resp.CertChain, nil
 }
 
-func (c *CitadelClient) getTLSDialOption() (grpc.DialOption, error) {
-	certPool, err := getRootCertificate(c.tlsOpts.RootCert)
-	if err != nil {
-		return nil, err
-	}
-	config := tls.Config{
-		GetClientCertificate: func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			var certificate tls.Certificate
-			key, cert := c.tlsOpts.Key, c.tlsOpts.Cert
-			if cert != "" {
-				var isExpired bool
-				isExpired, err = util.IsCertExpired(cert)
-				if err != nil {
-					citadelClientLog.Warnf("cannot parse the cert chain, using token instead: %v", err)
-					return &certificate, nil
-				}
-				if isExpired {
-					citadelClientLog.Warnf("cert expired, using token instead")
-					return &certificate, nil
-				}
-
-				// Load the certificate from disk
-				certificate, err = tls.LoadX509KeyPair(cert, key)
-				if err != nil {
-					return nil, err
-				}
-				c.usingMtls.Store(true)
```

Comment:
This seems lost

Reply:
Yes, i think this is the ut failure root cause

Reply:
Not related. During the debug, I found:

The reconnect logic happens when it is not using mtls. I doubt it is useless. The xds client does not have the reconnect logic.